<img src="http://meusite.mackenzie.br/rogerio/mackenzie_logo/UPM.2_horizontal_vermelho.jpg"  width=300, align="right">
<br>
<br>
<br>
<br>
<br>

# ***Projecão Imobiliária na Cidade de São Paulo: Um Modelo Preditivo de Precificação***
---

In [4]:
#@title **Identificação do Grupo**

#@markdown Integrantes do Grupo, nome completo em orgem alfabética (*informe \<TIA\>,\<nome\>*)
Aluno1 = '32139871, Júlia Ronquetti' #@param {type:"string"}
Aluno2 = 'None' #@param {type:"string"}
Aluno3 = 'None' #@param {type:"string"}
Aluno4 = 'None' #@param {type:"string"}
Aluno5 = 'None' #@param {type:"string"}



# **Resumo (*Abstract*)**

Como podemos prever com precisão o preço de um imóvel em um mercado tão variável? O projeto trata esse desafio utilizando uma abordagem que combina dados quantitativos, qualitativos de imóveis da cidade de São Paulo, coletadas por meio da técnica de *Web Scraping*. Os dados foram extraídos da plataforma OLX, sendo dados originais. A partir desses dados, diversos algoritmos de aprendizado de máquina serão testados para determinar o mais eficaz na previsão de preços.

# **Referencial Teórico**

### Implicações Práticas e Comerciais
  Modelos de previsão de preço de imóveis não ajudam somente compradores e vendedores nas suas situações pontuais, mas também a utilização de modelos avançados, como descritos pelo autor [Gampala et al. 2022] permite  aos investidores identificar oportunidades de compra e venda com maior precisão. Isso significa decisões de investimento mais informadas e potencialmente mais lucrativas. A precisão melhorada ajuda a prevenir riscos e maximizar retornos, tornando o investimento em imóveis uma opção mais atrativa.

  Além de melhorar as decisões de investimento, segundo o   [Anand et al. 2021], governos e autoridades locais podem utilizar os *insights* gerados por esses modelos para ajustar políticas de habitação e planejamento urbano. Isso inclui a determinação de áreas prioritárias para desenvolvimento, renovação ou mesmo para a implementação de controles de preço, evitando a especulação imobiliária e promovendo a acessibilidade habitacional.

### Fundamentos de Modelos Preditivos no Mercado Imobiliário
A avaliação de preços de imóveis no Brasil é orientada pela Associação Brasileira de Normas Técnicas (ABNT) que recomenda o uso de modelos econométricos para a avaliação de propriedades urbanas (NBR 14653-2 – “Avaliação de imóveis urbanos”)  [Associacão Brasileira de Normas Técnicas (ABNT) 2011]. Os avaliadores seguem essa recomendação coletando dados de imóveis semelhantes ao objetivo, como apartamentos residenciais na mesma cidade, e evoluindo o preço com características como área, localização, número de quartos, entre outros. O modelo estimado é então usado para determinar o valor de mercado do imóvel em questão.

Segundo o autor  [Marzagão et al. 2021], esse método ainda é muito utilizado, porém, é considerado falho, pois utiliza todas as amostras para ajustar a linha de regressão, sem deixar dados para testar o desempenho do modelo, o que não permite saber se os modelos criados são eficazes ou se são suposições aleatórias. Os autores argumentam que essa prática pode estabelecer grandes ineficiências no mercado, propondo em vez disso uma abordagem mais segura através de algoritmos de aprendizado de máquina, como árvores de decisão, máquinas de vetor de suporte e redes neurais, que podem descobrir padrões complexos nos dados e, potencialmente, oferecer predições mais precisas.

O avanço das técnicas de aprendizado de máquina oferece novas oportunidades para aprimorar a precisão das avaliações de preços de imóveis. Os autores [Gampala et al. 2022] enfatizam a importância de testar uma variedade de modelos preditivos para avaliar qual deles apresenta melhor desempenho, destacando que algoritmos avançados, quando adequadamente treinados e validados, podem proporcionar estimativas de preços mais confiáveis. Este processo inclui a validação cruzada e a análise comparativa das métricas de desempenho, garantindo que os modelos não apenas se ajustem bem aos dados de treinamento, mas também generalizem de forma eficaz para novos dados. Com base nisso, compradores e vendedores podem tomar decisões mais informadas e estratégicas, facilitando operações mais eficientes e lucrativas no mercado imobiliário.

### Web Scraping
O web scraping, também conhecido como extração de dados da web, é a técnica de coletar e processar dados automaticamente de websites. Essa técnica utiliza scripts ou ferramentas específicas para navegar em páginas da web, identificar e extrair informações relevantes, e armazená-las em um formato estruturado, como CSV, JSON ou banco de dados.

O web scraping é uma ferramenta poderosa para diversos fins, como Análise de mercado, na extração de dados sobre preços, produtos, tendências e concorrentes; pesquisa científica, na coleta de dados para estudos e análises quantitativas e qualitativas; monitoramento de websites, no rastreamento de alterações em preços, produtos ou informações; e automação de tarefas, na realização repetitiva de tarefas em websites, como downloads de arquivos ou envio de formulários.

Nesse trabalho, a técnica de web scraping foi utilizada para coletar dados de anúncios de imóveis no site OLX. Esses dados serão então utilizados para construir um modelo preditivo de preços de imóveis utilizando uma das técnicas de aprendizado de máquina mencionadas anteriormente. O modelo será avaliado em um conjunto de dados de validação para verificar sua precisão e generalização.

### Modelos de Machine Learning

A avaliação de imóveis é uma prática essencial no mercado imobiliário, sendo tradicionalmente realizada por meio de modelos de regressão linear. No entanto, essa abordagem tem se mostrado limitada e ineficiente devido à falta de testes de performance preditiva e à necessidade de hipóteses sobre a forma funcional do modelo [Marzagão et al., 2021]. Assim, o uso de algoritmos de aprendizado de máquina (*machine learning*) surge como uma alternativa promissora para melhorar a precisão e a eficiência das avaliações imobiliárias.

O estudo critica a dependência excessiva na regressão linear, destacando vários pontos fracos dessa abordagem. Primeiro, os modelos de regressão linear não são testados com dados separados, o que pode resultar em previsões pouco confiáveis [Marzagão et al., 2021]. Além disso, esses modelos exigem a especificação prévia de uma forma funcional, o que é inadequado devido à complexidade das interações entre as variáveis no mercado imobiliário [Marzagão et al., 2021]. Outro problema é o potencial de overfitting, onde um alto valor de R² pode não refletir um bom desempenho preditivo em novos dados [Marzagão et al., 2021].

# **Exemplo de Aplicação**

Uma possível aplicação seria criar uma ferramenta que permita a estimativa de preços de imóveis com maior precisão ao integrar diversas características relevantes dos imóveis mais recentes. Este projeto aborda a aplicação de técnicas de aprendizado de máquina para analisar e prever preços com base em variáveis como localização, área útil, número de quartos, banheiros, entre outros.

Neste contexto, um problema específico que será resolvido é a integração e a análise dos dados coletados através de web scraping da OLX. Utilizaremos técnicas de machine learning, como regressão linear e árvores de decisão, para construir e avaliar modelos preditivos que consigam capturar as tendências do mercado imobiliário com base nos dados coletados. A metodologia envolve a coleta e pré-processamento dos dados, a seleção e extração de features relevantes, o treinamento de modelos preditivos e a avaliação de sua performance.

É esperado um resultado relevante, e se esse não se mostrar eficaz, vale o estudo para entender como melhorar no futuro.

# **Implementação**

Coloque aqui o código da sua solução. Você deve explicar em linhas gerais o código. Para isso você pode quebrar o código em algumas partes e incluir células de texto explicativos. Não empregue para isso comentários no corpo do código, mas crie células de texto do notebook.

Tente empregar exemplos que não sejam muito complexos e que busquem dar a você ou ao leitor um ponto de partida para o uso da tecnologia/conceito explorado.

Ao final, se julgar útil, você pode ter uma célula com o código completo do que foi aqui dividido para explicar as diferentes partes.

O Web Scraping foi utilizado na plataforma OLX, porém não foi possível rodar aqui no colab, apenas no VS Code por conta de algumas bibliotecas que são utilizadas.
Segue o código abaixo:

## Web Scraping

Importa bibliotecas necessárias

In [ ]:
import json
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
import time
from bs4 import BeautifulSoup
import pandas as pd
import time
import json
import re
import warnings
warnings.filterwarnings('ignore')
import datetime
import math
pd.set_option('display.max_columns', 100)

Configura e inicializa o navegador, obtém o HTML da página e converte para BeautifulSoup

In [ ]:
def web_scrapping(url):
    chrome_service = Service(executable_path=r"./chromedriver.exe", log_path='NUL')
    chrome_options = webdriver.ChromeOptions()

    # opções para ignorar erros de certificado
    chrome_options.add_argument('--ignore-certificate-errors')
    chrome_options.add_argument('--allow-running-insecure-content')

    driver = webdriver.Chrome(service=chrome_service, options=chrome_options)
    driver.get(url)
    time.sleep(0.1)
    page_source = driver.page_source
    driver.quit()
    soup = BeautifulSoup(page_source, "html.parser")
    return soup


Extrai dados JSON do HTML

In [ ]:
def pegaJson(soup):
    # Encontrar o script com o ID '__NEXT_DATA__'
    next_data_script = soup.find('script', id='__NEXT_DATA__')

    # Verificar se o script foi encontrado
    if next_data_script:
        # Carregar o JSON contido no script
        json_data = json.loads(next_data_script.string)
        return json_data

    return None

Verifica e processa o HTML, extrai e estrutura os dados. Por mim, inclui em um dataframe

In [ ]:
def dataframe(soup):
    if soup:
        # Encontrar a tag <script> e extrair o JSON diretamente como um objeto Python
        data = pegaJson(soup)  # Agora 'data' já é um objeto Python (dict) retornado por 'pegaJson'
        if data:
            # Acessar os anúncios diretamente do objeto Python
            ads = data['props']['pageProps']['ads']
            dados_imoveis = []
            for imovel in ads:
                dados_imovel = {
                    "title": imovel.get("title", "Sem título"),
                    "price": imovel.get("price", "Sem Preço"),
                    "URL" : imovel.get("url", "Sem URL"),
                    "oldPrice": imovel.get("oldPrice", "Sem Preço"),
                    "location": imovel.get("location", "Sem localização"),
                    "date" : imovel.get("date", 0),
                    "imagem inicial" : imovel.get("thumbnail", "Sem imagem"),
                    "images": [img['original'] for img in imovel.get("images", [])],
                    "destaque" : imovel.get("isFeatured", "Sem informação"),
                }
                if 'properties' in imovel:
                    for prop in imovel["properties"]:
                        if prop["label"] == "Categoria":
                            dados_imovel["Categoria"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Tipo":
                            dados_imovel["Tipo"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Condomínio":
                            dados_imovel["Condomínio"] = prop.get("value", "Não informado")

                        elif prop["label"] == "IPTU":
                            dados_imovel["IPTU"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Área construída":
                            dados_imovel["Área construída"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Área útil":
                            dados_imovel["Área útil"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Quartos":
                            dados_imovel["Quartos"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Banheiros":
                            dados_imovel["Banheiros"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Vagas na garagem":
                            dados_imovel["Vagas na garagem"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Detalhes do imóvel":
                            dados_imovel["Detalhes do imóvel"] = prop.get("value", "Não informado")

                        elif prop["label"] == "Detalhes do condomínio":
                            dados_imovel["Detalhes do condomínio"] = prop.get("value", "Não informado")

                dados_imoveis.append(dados_imovel)
            df = pd.DataFrame(dados_imoveis)
            return df
        else:
            print("Não foi possível encontrar a tag <script> com o ID '__NEXT_DATA__'.")
    else:
        print("Não foi possível fazer o scraping da página.")


Coleta dados de múltiplas páginas e concatena os dataframes

In [ ]:
dfs = []
for page in range(1, 100):
    #Zona Oeste
    soup_zo = web_scrapping(f"https://www.olx.com.br/imoveis/venda/estado-sp/sao-paulo-e-regiao/zona-oeste?o={page}")
    df_zo = dataframe(soup_zo)
    df_zo['ZONA'] = 'OESTE'
    dfs.append(df_zo)

    #Zona Leste
    soup_zl = web_scrapping(f"https://www.olx.com.br/imoveis/venda/estado-sp/sao-paulo-e-regiao/zona-leste?o={page}")
    df_zl = dataframe(soup_zl)
    df_zl['ZONA'] = 'LESTE'
    dfs.append(df_zl)

    #Zona Sul
    soup_zs = web_scrapping(f"https://www.olx.com.br/imoveis/venda/estado-sp/sao-paulo-e-regiao/zona-sul?o={page}")
    df_zs = dataframe(soup_zs)
    df_zs['ZONA'] = 'SUL'
    dfs.append(df_zs)

    #Zona Norte
    soup_zn = web_scrapping(f"https://www.olx.com.br/imoveis/venda/estado-sp/sao-paulo-e-regiao/zona-norte?o={page}")
    df_zn = dataframe(soup_zn)
    df_zn['ZONA'] = 'NORTE'
    dfs.append(df_zn)

    #Zona Centro
    soup_zc = web_scrapping(f"https://www.olx.com.br/imoveis/venda/estado-sp/sao-paulo-e-regiao/centro?o={page}")
    df_zc = dataframe(soup_zc)
    df_zc['ZONA'] = 'CENTRO'
    dfs.append(df_zc)

df = pd.concat(dfs, ignore_index=True)

Tratamentos no dataframe

In [ ]:
# Convertendo a coluna 'images' que contém listas em strings
df['images'] = df['images'].apply(lambda x: str(x))
# Convertendo de volta para lista
# df['images'] = df['images'].apply(ast.literal_eval)
df_tratamentos = df.drop_duplicates()
df_tratamentos['date'] = df_tratamentos['date'].apply(lambda x: datetime.datetime.fromtimestamp(x, tz=datetime.timezone.utc).strftime('%d/%m/%Y'))
df_tratamentos['price'] = df_tratamentos['price'].replace({'R\$ ': '', '\.': ''}, regex=True)
df_tratamentos['oldPrice'] = df_tratamentos['oldPrice'].replace({'R\$ ': '', '\.': ''}, regex=True)
df_tratamentos['IPTU'] = df_tratamentos['IPTU'].replace({'R\$ ': '', '\.': ''}, regex=True)
df_tratamentos['Condomínio'] = df_tratamentos['Condomínio'].replace({'R\$ ': '', '\.': ''}, regex=True)
df_tratamentos['Área útil'].fillna(df_tratamentos['Área construída'], inplace=True)
df_tratamentos['Vagas na garagem'].fillna(0, inplace=True)
df_tratamentos.drop('Área construída', axis = 1, inplace=True)

df_tratamentos = df_tratamentos[(~df_tratamentos.price.isna())].reset_index(drop=True)
df_tratamentos = df_tratamentos[(~df_tratamentos['Área útil'].isna())].reset_index(drop=True)

df_tratamentos['Área útil'] = df_tratamentos['Área útil'].str.extract('(\d+)', expand=False)

imovel = df_tratamentos['Detalhes do imóvel'].str.split(',').explode().str.strip()
imovel = imovel.drop_duplicates().tolist()
imovel = [valor for valor in imovel if valor != "nan" and not (isinstance(valor, float) and math.isnan(valor))]

condominio = df_tratamentos['Detalhes do condomínio'].str.split(',').explode().str.strip()
condominio = condominio.drop_duplicates().tolist()
condominio = [valor for valor in condominio if valor != "nan" and not (isinstance(valor, float) and math.isnan(valor))]

def verificar_palavra(det, palavra):
    if pd.isnull(det) or not isinstance(det, str):
        return 0
    else:
        return 1 if palavra in det.split(', ') else 0

for palavra in condominio:
    df_tratamentos[palavra] = df_tratamentos['Detalhes do condomínio'].apply(lambda x: verificar_palavra(x, palavra))

for palavra in imovel:
    df_tratamentos[palavra] = df_tratamentos['Detalhes do imóvel'].apply(lambda x: verificar_palavra(x, palavra))

#Bairro
df_tratamentos['bairro'] = df_tratamentos['location'].str.split(', ').str[1]
df_tratamentos['bairro'] = df_tratamentos['bairro'].str.replace(r"\s*\(.*\)", "", regex=True)



## Alguns outros tratamentos e modelos


Como não foi possível rodar no colab, importa o dataframe

In [20]:
from google.colab import drive
drive.mount('/content/drive')
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 100)
import warnings
warnings.filterwarnings('ignore')
import os
import matplotlib.pyplot as plt
import gdown
import requests


from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.linear_model import LinearRegression, Ridge, Lasso
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor, GradientBoostingRegressor
from sklearn.svm import SVR
from sklearn.neighbors import KNeighborsRegressor
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [27]:
file_path = '/content/drive/MyDrive/IA Modelo preditivo/Dataframe_25_04.csv'
df = pd.read_csv(file_path, encoding = 'utf-8', sep=';')

Não deu certo ler direto de um arquivo publico no drive, portanto o link do dataframe está aqui: https://drive.google.com/file/d/19llLrQtT5EJTYWRskRA-JUf1EAjJ7dbB/view?usp=sharing

In [25]:
url = 'https://drive.google.com/file/d/19llLrQtT5EJTYWRskRA-JUf1EAjJ7dbB/view?usp=sharing'
r = requests.get(url, allow_redirects=True)

# Salve o arquivo
open('Dataframe_25_04.csv', 'wb').write(r.content)

# Carregue o DataFrame
df = pd.read_csv('Dataframe_25_04.csv', encoding='utf-8', sep=';')

ParserError: Error tokenizing data. C error: Expected 23 fields in line 59, saw 31


Realiza mais alguns tratamentos que foram necessários

In [4]:
campos_numericos = ['Condomínio', 'Área útil', 'Quartos', 'Banheiros', 'Vagas na garagem',
        'IPTU', 'Academia', 'Elevador', 'Permitido animais', 'Piscina', 'Portaria',
       'Salão de festas', 'Condomínio fechado', 'Segurança 24h',
       'Portão eletrônico', 'Área murada', 'Área de serviço',
       'Armários na cozinha', 'Armários no quarto', 'Churrasqueira',
       'Mobiliado', 'Quarto de serviço', 'Ar condicionado', 'Porteiro 24h',
       'Varanda']

#Ajustar o 5 ou mais nos campos: Quartos, Banheiros, Vagas na garagem
df['Quartos'] = np.where(df['Quartos'] == '5 ou mais', 5, df['Quartos'])
df['Banheiros'] = np.where(df['Banheiros'] == '5 ou mais', 5, df['Banheiros'])
df['Vagas na garagem'] = np.where(df['Vagas na garagem'] == '5 ou mais', 5, df['Vagas na garagem'])

#Fillna (0) nos campos numericos e transformar atributos object em int
df[campos_numericos] = df[campos_numericos].fillna(0).astype('int64')
#Ver o que fazer com os atributos de string


Seleciona colunas que serão usadas no modelo, realiza o one-hot encoding e separa a base em treinamento e teste

In [5]:
df_model = df[['price', 'Categoria', 'Tipo', 'destaque',
       'Condomínio', 'Área útil', 'Quartos', 'Banheiros', 'Vagas na garagem',
       'IPTU', 'ZONA',
       'Academia', 'Elevador', 'Permitido animais', 'Piscina', 'Portaria',
       'Salão de festas', 'Condomínio fechado', 'Segurança 24h',
       'Portão eletrônico', 'Área murada', 'Área de serviço',
       'Armários na cozinha', 'Armários no quarto', 'Churrasqueira',
       'Mobiliado', 'Quarto de serviço', 'Ar condicionado', 'Porteiro 24h',
       'Varanda', 'bairro']]

df_model = pd.get_dummies(df_model, columns=['Categoria', 'Tipo', 'ZONA', 'bairro'], drop_first=True)
X = df_model.drop('price', axis = 1)
y = df_model.price
X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.25, random_state=123)

Treina diversos modelos e compara algumas métricas de avaliação

In [6]:
modelos = {
    'Linear Regression': LinearRegression(),
    'Ridge Regression': Ridge(),
    'Lasso Regression': Lasso(),
    'Decision Tree': DecisionTreeRegressor(),
    'Random Forest': RandomForestRegressor(),
    'Gradient Boosting': GradientBoostingRegressor(),
    'Support Vector Regressor': SVR(),
    'K-Nearest Neighbors': KNeighborsRegressor()
}

# Avaliar cada modelo
resultados = {}
for nome, modelo in modelos.items():
    modelo.fit(X_train, Y_train)
    y_pred = modelo.predict(X_test)
    mse = mean_squared_error(Y_test, y_pred)
    mae = mean_absolute_error(Y_test, y_pred)
    r2 = r2_score(Y_test, y_pred)
    resultados[nome] = {'MSE': mse, 'MAE': mae, 'R²': r2}

# Exibir os resultados
resultados_df = pd.DataFrame(resultados).T
print(resultados_df)

                                   MSE            MAE        R²
Linear Regression         1.571225e+12  601451.958935  0.527870
Ridge Regression          1.573476e+12  598216.613699  0.527194
Lasso Regression          1.570121e+12  600876.543281  0.528202
Decision Tree             1.153384e+12  340805.202673  0.653425
Random Forest             6.968856e+11  275396.454885  0.790596
Gradient Boosting         8.949052e+11  410264.869885  0.731094
Support Vector Regressor  3.577487e+12  876796.178874 -0.074981
K-Nearest Neighbors       1.265267e+12  422158.295853  0.619806


O **Random Forest** apresenta o menor MSE e MAE, além de ter o maior valor de R², o que indica que é o modelo que melhor se ajusta aos dados e fornece as previsões mais precisas.

Já o **Support Vector Regressor** apresenta o maior MSE e MAE, além de ter o menor valor de R², indicando que é o modelo que pior se ajusta aos dados e fornece as previsões menos precisas.

# **Uma ideia de Projeto de Aplicação**

Com a crescente disponibilidade de dados e o avanço das tecnologias de aprendizado de máquina, há uma oportunidade de desenvolver um sistema automatizado e atualizado frequentemente que possa fornecer avaliações de preços mais precisas e consistentes, ajudando a mitigar a subjetividade e as variações.

Uma ideia de projeto seria dsenvolver um sistema inteligente de avaliação de preços de imóveis que utilize uma combinação de dados quantitativos e qualitativos. Este sistema empregará técnicas de aprendizado de máquina para analisar dados coletados de várias fontes, como sites de anúncios de imóveis, registros públicos e dados socioeconômicos. Esse modelo poderá ser disponibilizado também pelos próprios sites de divulgação de imóveis.

Este sistema automatizado de avaliação de preços de imóveis pode trazer maior transparência e eficiência ao mercado imobiliário, fornecendo estimativas de preços mais precisas e baseadas em dados objetivos. Isso pode ajudar compradores a tomar decisões informadas e vendedores a definir preços competitivos, contribuindo para um mercado mais justo e equilibrado.


# **Referências**

Associacão Brasileira de Normas Técnicas (ABNT) (2011). Avaliacão de bens
parte 2: Imóveis urbanos. Norma Técnica NBR 14653-2. Disponível em:
https://www.prefeitura.sp.gov.br/cidade/secretarias/upload/infraestrutura/arquivos/ACESSO

Marzagão, T., Ferreira, R., and Sales, L. (2021). A note on real estate appraisal in brazil.
Revista Brasileira de Economia, 75(1):29–36.

Gampala, V., Sai, N. Y., and Sai Bhavya, T. N. (2022). Real-estate price prediction sys-
tem using machine learning. In 2022 International Conference on Applied Artificial
Intelligence and Computing (ICAAIC), pages 533–538

S. P. Sreeja, V. Asha, B. Saju, N. CP, P. O. Prakash and A. K. Singh, "Real Estate Price Prediction using Machine Learning," 2023 Third International Conference on Advances in Electrical, Computing, Communication and Sustainable Technologies (ICAECT), Bhilai, India, 2023, pp. 1-7, doi: 10.1109/ICAECT57570.2023.10117910.




# **Vídeo e GitHub**

https://youtu.be/hCQDEaEz62U

# **Conclusão**

Este projeto de desenvolvimento de um sistema de avaliação de preços de imóveis apresenta uma abordagem baseada em dados originais para enfrentar um problema relevante e atual no mercado imobiliário. Utilizando técnicas de aprendizado de máquina o sistema proposto visa fornecer estimativas de preços consistentes, mitigando a subjetividade e a variação inerentes aos métodos tradicionais.

Porém o modelo chegou em um resultado de R2 de 79% e com um MAE de 275396.45 usando o algoritmo Random Forest. Acredito que ainda conseguimos melhorar esse resultado. E um possível avanço para um resultado melhor poderia ser a utilização de imagens desses imóveis. No futuro, uma extensão promissora deste trabalho é a inclusão de imagens dos imóveis no modelo preditivo. Utilizando técnicas de visão computacional e deep learning, como redes neurais convolucionais (CNNs), será possível extrair características visuais significativas das imagens dos imóveis que impactam diretamente seu valor de mercado. A integração dessas características visuais com os dados quantitativos e qualitativos poderá aprimorar ainda mais a precisão das previsões, oferecendo um sistema de avaliação de preços ainda mais robusto e confiável.

O autor realizou um projeto similar e alcançou um MAE de R$35,463.24, utilizando o mesmo algoritmo. Portanto, será interessante analizar como diminuir o MAE que alcancei.



# **Apêndice**

Link da Apresentação no YouTube:


In [31]:
from IPython.display import YouTubeVideo
YouTubeVideo('https://youtu.be/hCQDEaEz62U')
#https://youtu.be/hCQDEaEz62U

---

In [2]:
#@title **Avaliação**
Referencial_Teorico = 4 #@param {type:"slider", min:0, max:10, step:1}

Conceitos_Chave = 9 #@param {type:"slider", min:0, max:10, step:1}

Exemplo_Aplicacao = 6 #@param {type:"slider", min:0, max:10, step:1}

Ideia_Projeto = 7 #@param {type:"slider", min:0, max:10, step:1}

Conclusao = 7 #@param {type:"slider", min:0, max:10, step:1}








In [5]:
#@title **Nota Final**
nota = Referencial_Teorico + Conceitos_Chave + 2*Exemplo_Aplicacao + 2*Ideia_Projeto + Conclusao

nota = nota / 7

print(f'Nota final do trabalho {nota :.1f}')

import numpy as np
import pandas as pd

alunos = pd.DataFrame()

lista_tia = []
lista_nome = []

for i in range(1,6):
  exec("if Aluno" + str(i) + " !='None':  lista = Aluno" + str(i) + ".split(','); lista_tia.append(lista[0]); lista_nome.append(lista[1].upper())")

alunos['tia'] = lista_tia
alunos['nome'] = lista_nome
alunos['nota'] = np.round(nota,1)
print()
display(alunos)

Nota final do trabalho 7.4



tia              nome  nota
0  32139871   JÚLIA RONQUETTI   7.4